### ISSUE TITLE
mailroom-ml: Modal serve fallback KeyErrors on `unknown` doc_type instead of abstaining

### ISSUE THREAD

**Issue by Exios66** [LLM-Mailroom-Services/mailroom-issues#121]
**Repository:** mailroom-ml
**Package:** —
**Severity:** medium
**Found:** 2026-09-21 repo-wide read-only scan (code-analyst)

`predict` indexes `logits_by_head[f"logits_{dt['label']}"]` unconditionally (`deploy/serve_app.py:252-253`), but the doc_type label space includes `unknown` (`src/mailroom_ml/labels.py:281`), which has **no subclass head** (`training/train_modernbert.py:1308`; guards at `src/mailroom_ml/inference.py:480,521,798`). When the doc_type argmax lands on `unknown`, the fallback endpoint raises KeyError → HTTP 500 instead of abstaining/routing.

**Fix:** Mirror the inference guard — if the label has no head (or is `unknown`), return `subclass: null` with an abstain / routed-to-LLM marker.
**Confidence:** HIGH, EVIDENCED (frequency of `unknown` winning argmax: INFERRED — untrained row, rare)
**Dedupe:** none found (family: #43 abstention design)
**Related:** `deploy/serve_app.py:252`; `src/mailroom_ml/inference.py:480-521`

**Comment by Exios66** [comment]
### Verification
- Verdict: **VERIFIED_TRUE**
- Evidence:
  - `deploy/serve_app.py` L252–253 (pre-fix): unconditional `logits_{dt['label']}` — `unknown` doc_type has no subclass head (`src/mailroom_ml/inference.py` guards ~L480–521).
  - `src/mailroom_ml/labels.py` — `unknown` in doc_type label space.

### Fix (if applicable)
- PR: https://github.com/LLM-Mailroom-Services/mailroom-ml/pull/3
- What changed:
  - `deploy/serve_app.py` — `_subclass_prediction()` mirrors inference abstain when label is `unknown` or head missing.
  - `tests/test_deploy.py` — `test_serve_subclass_abstains_on_unknown_doc_type`.
- Contract/test proof:
  ```bash
  cd mailroom-ml && pip install fastapi modal -q && python3 -m pytest tests/test_deploy.py::test_serve_subclass_abstains_on_unknown_doc_type -q
  ```
  **1 passed** (cloud agent, 2026-09-24).

### Close policy
- Leave **OPEN** until PR #3 merges.

**Comment by Exios66** [comment]
## Final proof (post-merge)

**Verdict:** FIXED_ON_MAIN
**Fix PR:** https://github.com/LLM-Mailroom-Services/mailroom-ml/pull/3
**Squash merge SHA:** `ed4bf204c93145444a914904d2054cad9371b0a4`
**Destination repo:** LLM-Mailroom-Services/mailroom-ml

Jack authorized merge of the medium verify→fix batch. Fix is on `main`. Closing as completed per proof-on-close contract.
